components/hooks/useCountdownTime.ts

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from "react";
+
+interface CurrentPrevious {
+  current: Countdown;
+  previous: Countdown;
+}
+
+interface Countdown {
+  days: number;
+  hours: number;
+  minutes: number;
+  seconds: number;
+}
+
+function getTimeLeft(endDate: Date) {
+  const now = new Date().getTime();
+  const distance = endDate.getTime() - now;
```

Comment:
seems like it allows distance < 0 ，which would make all the number negative.

we can just add a distance = max(endDate.getTime() - now,0) here to prevent the countdown being broken